src/transfer.ts

```diff
@@ -25,6 +25,45 @@ export const transferStatusSchema = z.nativeEnum(TransferStatus, {
   description: 'transferStatusSchema',
 })
 
+// These enums are first segmented between transfers in and transfers out
+// since they both allow a slightly different set of statuses.
+// Transfer in statuses are further segmented, since once a transfer in
+// progresses to sending the user funds, providers are expected
+// to return new metadata about the transfer. This segmentation allows
+// us to encode those semantics into Zod schemas.
+export enum TransferInStatusPreTx {
+  TransferStarted = TransferStatus.TransferStarted,
+  TransferFiatFundsDebited = TransferStatus.TransferFiatFundsDebited,
+  TransferReceivedFiatFunds = TransferStatus.TransferReceivedFiatFunds,
+  TransferFailed = TransferStatus.TransferFailed,
+}
+export const transferInStatusPreTxSchema = z.nativeEnum(TransferInStatusPreTx, {
```

Comment:
```suggestion
export const transferInStatusPreTxSchema = z.enum(
  [
    TransferStatus.TransferStarted,
    TransferStatus.TransferFiatFundsDebited,
    TransferStatus.TransferReceivedFiatFunds,
    TransferStatus.TransferFailed,
  ],
```

Reply:
I found the `TransferInStatusPreTx` and native enums created in this PR kinda clunky, so I tried out a different approach. I think the downside is that if you actually wanted a typescript enum for TransferInStatusPreTx, you wouldn't have one, but I suspect the only actual use case for it is for typing, which you can get from the zod schema.

Reply:
feel free to adopt this approach or stick with what you've got